src/redux/request.js

```diff
@@ -0,0 +1,271 @@
+import queryString from 'query-string';
+import startsWith from 'lodash/startsWith';
+import { handleActions as reduxHandleActions } from 'redux-actions';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/observable/from';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/operator/catch';
+
+// action types
+export const REQUEST_MAKE = '@@ui-eholdings/REQUEST_MAKE';
+export const REQUEST_RESOLVE = '@@ui-eholdings/REQUEST_RESOLVE';
+export const REQUEST_REJECT = '@@ui-eholdings/REQUEST_REJECT';
+
+// action creators
+
+/**
+ * Creates an action to start a request for a specified named epic
+ *
+ * @param {string} name - name of the epic to trigger the request
+ * @param {object} [data={}] - data to provide to the request (query
+ * params for GET/DELETE; body data for PUT/POST)
+ * @param {object} [options={}] - additional options for this request
+ * (currently only `method` is supported)
+ */
+export const makeRequest = (name, data = {}, options = {}) => ({
+  type: REQUEST_MAKE,
+  name,
+  data,
+  options
+});
+
+export const resolveRequest = (name, payload) => ({
+  type: REQUEST_RESOLVE,
+  payload,
+  name
+});
+
+export const rejectRequest = (name, error) => ({
+  type: REQUEST_REJECT,
+  error,
+  name
+});
+
+/**
+ * Request reducer creator
+ *
+ * Creates a reducer that responds to three basic request actions to
+ * provide promise-like state properties for the request.
+ *
+ * - REQUEST_MAKE sets `isPending` to `true` and reinits the default
+ * promise-like properties.
+ * - REQUEST_RESOLVE sets `isPending` to `false`, `isResolved` to
+ * `true`, and `content` to the action's payload.
+ * - REQUEST_REJECT sets `isPending` to `false`, `isRejected` to
+ * `true`, and `error` to the action's error property.
+ *
+ * In order for request reducers to receive actions from a specific
+ * request epic, the names of each reducer/epic combo must match
+ * exactly. Additonally, it must be unique to a reducer/epic combo,
+ * otherwise other epics with the same name will trigger the reducer.
+ *
+ * if `handleActions` is provided, other actions can be included in
+ * the reducer to further alter it's state. When one of the request
+ * actions are provided, the resulting state is reduced again by the
+ * default request action handler to maintain the promise-like
+ * properties of the state.
```

Comment:
An example of this would be awesome. Also, something indicating which reduction happens _first_